server/service/src/sync/sync_on_central/mod.rs

```diff
@@ -0,0 +1,116 @@
+use repository::{ChangelogFilter, ChangelogRepository, ChangelogTableName, EqualFilter};
+
+use util::{format_error, is_central_server};
+
+use crate::{
+    service_provider::ServiceProvider,
+    sync::{api::SyncApiV5, translations::PushTranslationType},
+};
+
+use super::{
+    api_v6::{SyncBatchV6, SyncParsedErrorV6, SyncRecordV6, SyncRequestV6},
+    translations::translate_changelogs_to_push_records,
+};
+
+fn create_filter() -> ChangelogFilter {
+    ChangelogFilter::new().table_name(EqualFilter {
+        equal_any: Some(vec![ChangelogTableName::PackVariant]),
+        ..Default::default()
+    })
+
+    // TODO, the idea for this method is to build a query in such a way as to allow
+    // extracting all relevant records for a site from change_log, where resulting SQL would be
+    // SELECT * FROM changelog_dedup
+    // WHERE cursor > {remote site SyncPullCursorV6} AND last_sync_site_id != {remote site id}
+    // AND
+    // (
+    // 	table_name in {central_record_names}
+    //  OR
+    // 	(table_name in {transfer record names}  AND name_id IN {name_ids of active stores on remote site})
+    //  OR
+    // 	// Special cases
+    // 	(table_name in {patient record name} AND patient_id IN {select name_id from name_store_join where store_id in {active stores on remote site})
+    // )
+    // When we upgrade to diesel 2 we can do dynaimc filter: https://github.com/andreievg/diesel-rs-dynamic-filters
+    // And the above would become something like:
+    // use ChangeLog::Filter as f;
+    // let filter = create_and_filter(vec![
+    //     f::cursor(NumberFilter::GreaterThen(cursor)),
+    //     f::last_sync_site_id(NumberFilter::NotEqual(site_id)),
+    //     f::Or(
+    //         vec![f::table_name(StringFilter::Any(central_record_table_names))],
+    //         f::And(vec![
+    //             f::table_name(StringFilter::Any(transfer_record_names)),
+    //             f::name_id(StringFilter::Any(active_stores_on_site.name_ids())),
+    //         ]),
+    //         f::And(vec![
+    //             f::table_name(StringFilter::Any(remote_record_names)),
+    //             f::store_id(StringFilter::Any(active_stores_on_site.store_ids())),
+    //         ]),
+    //         f::And(vec![
+    //             f::table_name(StringFilter::Any(patient_record_names)),
+    //             f::patient_id(StringFilter::Any(
+    //                 name_store_joins::create_filtered_query(name_store_join::Filter::store_id(
+    //                     StringFitler::Any(active_stores_on_site.store_ids()),
+    //                 ))
+    //                 .select(name_store_joins::name_id),
+    //             )),
+    //         ]),
+    //         // Etc..
+    //     ),
+    // ]);
+}
+
+pub async fn pull(
+    service_provider: &ServiceProvider,
+    SyncRequestV6 {
+        cursor,
+        batch_size,
+        sync_v5_settings,
+    }: SyncRequestV6,
+) -> Result<SyncBatchV6, SyncParsedErrorV6> {
+    use SyncParsedErrorV6 as Error;
```

Comment:
Why not just  `use SyncParsedErrorV6` personally not a big fan of this re-naming pattern. I guess it makes copy and paste easier?

Reply:
> I guess it makes copy and paste easier

That and less verbose down the line, especially when used in chain mapping with `map_err` (most likely the latter)